llvm/lib/Transforms/Instrumentation/BoundsChecking.cpp

```diff
@@ -108,14 +108,17 @@ static Value *getBoundsCheckCond(Value *Ptr, Value *InstVal,
   return Or;
 }
 
-static CallInst *InsertTrap(BuilderTy &IRB, bool DebugTrapBB) {
+static CallInst *InsertTrap(BuilderTy &IRB, bool DebugTrapBB,
+                            std::optional<int8_t> GuardKind) {
   if (!DebugTrapBB)
```

Comment:
this applies only only !DebugTrapBB ?

Reply:
There is no precedent in CGExpr.cpp on adding the GuardKind to `Intrinsic::trap` for UBSan, hence this patch only applies to `DebugTrapBB` / `Intrinsic::ubsantrap`.

Reply:
Never-mind, I confused DebugTrapBB with SingleTrapBB